sysdig/resource_sysdig_monitor_alert_v2_common.go

```diff
@@ -180,36 +211,50 @@ func buildAlertV2CommonStruct(d *schema.ResourceData) (alert *monitor.AlertV2Com
 
 		for _, channel := range attr.(*schema.Set).List() {
 			channelMap := channel.(map[string]interface{})
+			channelID := channelMap["id"].(int)
+			nc, err := client.GetNotificationChannelById(ctx, channelID)
+			if err != nil {
+				return nil, fmt.Errorf("error getting info for notification channel %d: %w", channelID, err)
+			}
 			newChannel := monitor.NotificationChannelConfigV2{
-				ChannelID: channelMap["id"].(int),
-				Type:      channelMap["type"].(string),
+				ChannelID: channelID,
+				Type:      nc.Type,
 			}
 
 			if renotifyEveryMinutes, ok := channelMap["renotify_every_minutes"]; ok {
-				newChannel.Options = &monitor.NotificationChannelOptionsV2{
-					ReNotifyEverySec: minutesToSeconds(renotifyEveryMinutes.(int)),
+				m := renotifyEveryMinutes.(int)
+				if m != 0 {
+					s := minutesToSeconds(m)
+					newChannel.OverrideOptions.ReNotifyEverySec = &s
 				}
 			}
+
+			newChannel.OverrideOptions.Thresholds = []string{}
+			main_threshold := channelMap["main_threshold"].(bool)
+			if main_threshold {
+				newChannel.OverrideOptions.Thresholds = append(newChannel.OverrideOptions.Thresholds, "MAIN")
+			}
+			warning_threshold := channelMap["warning_threshold"].(bool)
+			if warning_threshold {
+				newChannel.OverrideOptions.Thresholds = append(newChannel.OverrideOptions.Thresholds, "WARNING")
+			}
+
 			channels = append(channels, newChannel)
 		}
 		alert.NotificationChannelConfigList = &channels
 	}
 
+	customNotification := monitor.CustomNotificationTemplateV2{}
```

Comment:
I remember finding the APIs setting automatically the `defaultTitle` when the customNotification was sent empty, that's the reason behind the weird logic that was there before. Is this not the case anymore?